src/main/java/de/muenchen/allg/itd51/wollmux/db/DataFinder.java

```diff
@@ -131,41 +90,40 @@ protected int find(String dbSpalte, String value)
 
   /**
    * Sucht in der Datenquelle nach Datensätzen wobei die beiden Suchbedingungen
-   * (dbSpalte1==evaluate(value1) und dbSpalte2==evaluate(value2)) mit UND
-   * verknüpft sind - die gefundenen Werte werden danach in die PAL kopiert.
+   * (dbSpalte1==evaluate(value1) und dbSpalte2==evaluate(value2)) mit UND verknüpft sind - die
+   * gefundenen Werte werden danach in die PAL kopiert.
    *
-   * @param dbSpalte1
-   *          der Feldname über den nach dem evaluierten Wert von value gesucht
-   *          wird.
-   * @param value1
-   *          value wird vor der Suche mittels evaluate() evaluiert (d.h. evtl.
-   *          vorhandene Variablen durch die entsprechenden Inhalte ersetzt
-   *          ersetzt).
-   * @param dbSpalte2
-   *          der Feldname über den nach dem evaluierten Wert von value gesucht
-   *          wird.
-   * @param value2
-   *          value wird vor der Suche mittels evaluate() evaluiert (d.h. evtl.
-   *          vorhandene Variablen durch die entsprechenden Inhalte ersetzt
-   *          ersetzt).
+   * @param query
+   *          value0: der Feldname über den nach dem evaluierten Wert von value gesucht wird.
+   *          value1: wird vor der Suche mittels evaluate() evaluiert (d.h. evtl. vorhandene
+   *          Variablen durch die entsprechenden Inhalte ersetzt ersetzt).
+   * @param query2
+   *          value0: der Feldname über den nach dem evaluierten Wert von value gesucht wird.
+   *          value1: wird vor der Suche mittels evaluate() evaluiert (d.h. evtl. vorhandene
+   *          Variablen durch die entsprechenden Inhalte ersetzt ersetzt).
    * @return die Anzahl der gefundenen Datensätze
    */
-  public int find(String dbSpalte1, String value1, String dbSpalte2,
-      String value2)
+  public int find(Pair<String, String> query, Pair<String, String>... query2)
```

Comment:
Either query2 is a single pair than it's all fine, but if it is an array what about the other elements (index above 0)?